### PR TITLE
[Merged by Bors] - chore: split Analysis.Convex.Cone.Basic

### PR DESCRIPTION
**Pull request by kim-em**
Splits `Mathlib.Analysis.Convex.Cone.Basic`, to move Riesz extension and Hahn-Banach out of the basic file about definitions.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Analysis/Convex/Cone/Extension.lean [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+* `ConvexCone.hyperplane_separation_of_nonempty_of_isClosed_of_nmem`:
+  This variant of the
+  [hyperplane separation theorem](https://en.wikipedia.org/wiki/Hyperplane_separation_theorem)
+  states that given a nonempty, closed, convex cone `K` in a complete, real inner product space `H`
+  and a point `b` disjoint from it, there is a vector `y` which separates `b` from `K` in the sense
+  that for all points `x` in `K`, `0 ≤ ⟪x, y⟫_ℝ` and `⟪y, b⟫_ℝ < 0`. This is also a geometric
+  interpretation of the
+  [Farkas lemma](https://en.wikipedia.org/wiki/Farkas%27_lemma#Geometric_interpretation).
+* `ConvexCone.inner_dual_cone_of_inner_dual_cone_eq_self`:
+
```
Comment (apurvnakade) [inline Mathlib/Analysis/Convex/Cone/Extension.lean:39]:
```suggestion
```

File: Mathlib/Analysis/Convex/Cone/Dual.lean [hunk 2]
```diff
@@ -22,7 +22,6 @@ We prove the following theorems:
```
Comment (apurvnakade) [inline Mathlib/Analysis/Convex/Cone/Dual.lean:22]:
```suggestion
* `ConvexCone.hyperplane_separation_of_nonempty_of_isClosed_of_nmem`:
  This variant of the
  [hyperplane separation theorem](https://en.wikipedia.org/wiki/Hyperplane_separation_theorem)
  states that given a nonempty, closed, convex cone `K` in a complete, real inner product space `H`
  and a point `b` disjoint from it, there is a vector `y` which separates `b` from `K` in the sense
  that for all points `x` in `K`, `0 ≤ ⟪x, y⟫_ℝ` and `⟪y, b⟫_ℝ < 0`. This is also a geometric
  interpretation of the
  [Farkas lemma](https://en.wikipedia.org/wiki/Farkas%27_lemma#Geometric_interpretation).
```

**Review by apurvnakade — APPROVED** [review]
This is a nice idea! These two theorems seem too big for `Basic.lean`

The comments at the start of the files seem to be incorrectly arranged after the refactor. But otherwise looks good.

**Comment by j-loreaux** [comment]
Thanks! Definitely a worthwhile split.

bors merge

**Pull request closed without merging**